src/enzyme_ad/jax/Passes/LLVMToAffineAccess.cpp

```diff
@@ -300,6 +301,40 @@ static MemRefVal convertToMemref(PtrVal addr) {
                                  addr.getType().getAddressSpace());
   // TODO we can actually plug in the size of the memref here if `addr` is
   // defined by an llvm.alloca
+
+  // Return with exact shape and size if
+  // 1. addr is a gep
+  // 2. the base addr was constructed using Memref2Pointer
+  if (auto gep = addr.getDefiningOp<LLVM::GEPOp>()) {
+    auto basePtr = gep.getBase();
+
+    // if basePtr is from a memref, then emit a static shaped Pointer2memref
+    if (auto m2p = basePtr.getDefiningOp<enzymexla::Memref2PointerOp>()) {
+      // get the original memref + type
+      Value origMemref = m2p.getSource();
+      auto origMemrefType = cast<MemRefType>(origMemref.getType());
+
+      // collapse dimensions
```

Comment:
see the delinearize pass which I imagine is the right place to add support for multidim of memref load/store

Reply:
Let me take a look - but is there a case where collapsing dimensions wouldn't work? I'm assuming it's when you have multiple GEP indices (which I haven't seen in Oceananigans)

Reply:
there's a distinction from the memref being 1d and the gep being 1d. We should support memref of any size, its fine to assume the gep is 1d for now